### ISSUE TITLE
How to adjust the parameters correctly ?

### ISSUE THREAD

**Issue by marc40000** [niuzhiheng/caffe#6]
I'm not sure if this can be answered easily. I created two small level-dbs containing different images of bananas and keyboards. Both contain 10 images, 256^2 pixels in size. I calculated the mean of them and modified imagenet__.prototxt to refernce those files. I did the same with the lenet__.prototxt files. When I run train_net on those, it absically seem to work, but super slow. The mnist dataset finishes in about 30 minutes, but the same prototxt files referencing my dbs take 1.2 h to display only the first test score:

I0621 12:19:46.140650  1944 solver.cpp:106] Iteration 0, Testing net
I0621 13:30:20.244161  1944 solver.cpp:142] Test score #0: 0
I0621 13:30:20.245159  1944 solver.cpp:142] Test score #1: 2.30191

Using the imagenet-prototxt files, nothing happens for the first 1.5 h.

Am I doing something completely wrong? This code was used to train imagenet images, so my 20 images alone can't make it so slow?

**Comment by marc40000** [comment]
6 h later, the output of the adapted lenet-prototxt files looks like this:

I0621 12:19:46.140650  1944 solver.cpp:106] Iteration 0, Testing net
I0621 13:30:20.244161  1944 solver.cpp:142] Test score #0: 0
I0621 13:30:20.245159  1944 solver.cpp:142] Test score #1: 2.30191
I0621 16:12:01.778450  1944 solver.cpp:237] Iteration 100, lr = 0.00992565
I0621 16:12:02.152473  1944 solver.cpp:87] Iteration 100, loss = -1.#IND
I0621 18:29:29.144794  1944 solver.cpp:237] Iteration 200, lr = 0.00985258
I0621 18:29:29.461815  1944 solver.cpp:87] Iteration 200, loss = -1.#IND

loss = -1.#IND, is that ok or is something wrong?

**Comment by marc40000** [comment]
The image net based one also outputted something now

I0621 12:19:52.768038 11272 solver.cpp:61] Solving CaffeNet
I0621 12:19:52.768038 11272 solver.cpp:106] Iteration 0, Testing net
I0621 19:33:09.547161 11272 solver.cpp:142] Test score #0: 0
I0621 19:33:09.548122 11272 solver.cpp:142] Test score #1: 7.57604
I0621 21:53:30.233541 11272 solver.cpp:237] Iteration 20, lr = 0.01
I0621 21:53:30.496578 11272 solver.cpp:87] Iteration 20, loss = 2.23954
I0621 23:53:55.748026 11272 solver.cpp:237] Iteration 40, lr = 0.01
I0621 23:53:56.336067 11272 solver.cpp:87] Iteration 40, loss = -1.#IND

Also "loss = -1.#IND". Is that ok?

**Comment by niuzhiheng** [comment]
You'd like to refer to this [issue](https://github.com/BVLC/caffe/issues/373) from the official Caffe. Normally, this is caused by a improper learning rate.
